content/3.architecture/2.failure-scenarios.md

```diff
@@ -0,0 +1,22 @@
+# Failure Scenarios
+
+## Overview
+
+This document outlines the different failure scenarios we have anticipated and what our mitigation is.
+
+
+## Failure Scenarios
+
+| Event             | Consequences                                   | Mitigations                                        |
+|------------------ | -----------------------------------------------| ---------------------------------------------------|
+| Github Failure    | Github is a dependency for Flux. Without Github _changes_ to the manifests cannot be applied, including the deployment of new applications, infrastructure change. It doesn't affect the operational state of the cluster. If using Github as package (image) registry as well, new applications that have to run won't be able to do. | - |
+| Cloud  Failure    | In case of complete Cloud failure, our platform will be unaccessible (partially or fully). | Choose trustworthy provider. A costly mitigation is having a hot side in another provider. Given the scope of our operations, this is **not** recommended |
+| Storage Failure   | Object storage within the cloud should be used mostly for non-operational tasks, such as backups. If used for stateful applications, then unavailability of this will cause the corresponding applications to fail. | - |
```

Comment:
one could argue cloud storage failure (like backblaze B2) while damaging, could be less likely that failure of internal storage solutions

Reply:
I definitely agree, although we still need to consider the eventuality. In case of external providers we need to keep into account that failure can also be a simple network glitch (our cloud, their side or some ISP in the middle). But yeah, in general the risk is fairly low.